src/lib/MetamaskDelegatedAccess.ts

```diff
@@ -0,0 +1,82 @@
+import { Wallet } from "ethers"
+import { Buffer } from 'buffer'
+import { encrypt } from "eth-sig-util"
+import { bufferToHex } from 'ethereumjs-util'
+import {StreamrClient} from "streamr-client"
+
+declare global {
+    interface Window { 
+        ethereum: any
+    }
+}
+
+export class MetamaskDelegatedAccess {
+    metamaskAddress?: string 
+    clientAddress?: string
+
+    constructor(){
+    }
+
+    public async connect(): Promise<{client: StreamrClient, address: string}>{
+        // enable and fetch metamask account
+        await window.ethereum.enable()
```

Comment:
We should protect against the use case of a nocoiner trying this out. 
I've made a ticket for that - https://linear.app/streamr/issue/FRONT-736/app-shouldnt-break-when-theres-no-metamask probs better to merge this and work on that next.